sylt-compiler/src/typechecker.rs

```diff
@@ -1137,9 +1246,44 @@ impl TypeChecker {
             return *res;
         }
         let new_ty = self.push_type(Type::Unknown);
-        self.find_node_mut(new_ty).constraints = self.find_node(old_ty).constraints.clone();
         seen.insert(old_ty, new_ty);
 
+        self.find_node_mut(new_ty).constraints = self
+            .find_node(old_ty)
+            .constraints
+            .clone()
```

Comment:
This clone feels reduntant. Constraints could be `Copy` or the iterator could be cloned.